packages/realm-web/src/UserProfile.ts

```diff
@@ -65,79 +67,63 @@ const DATA_MAPPING: { [k in DataKey]: keyof UserProfile } = {
 };
 
 /** @inheritdoc */
-export class UserProfile implements Realm.UserProfile {
-    /** @inheritdoc */
-    public readonly name?: string;
-
-    /** @inheritdoc */
-    public readonly email?: string;
-
-    /** @inheritdoc */
-    public readonly pictureUrl?: string;
-
-    /** @inheritdoc */
-    public readonly firstName?: string;
-
-    /** @inheritdoc */
-    public readonly lastName?: string;
-
-    /** @inheritdoc */
-    public readonly gender?: string;
-
-    /** @inheritdoc */
-    public readonly birthday?: string;
-
-    /** @inheritdoc */
-    public readonly minAge?: string;
-
-    /** @inheritdoc */
-    public readonly maxAge?: string;
-
-    /** @inheritdoc */
+export class UserProfile<UserProfileDataType = Realm.DefaultUserProfileData> {
+    /** @ignore */
     public readonly type: Realm.UserType = UserType.Normal;
 
-    /** @inheritdoc */
+    /** @ignore */
     public readonly identities: Realm.UserIdentity[] = [];
 
+    /** @ignore */
+    public readonly data: UserProfileDataType;
+
     /**
      * @param response The response of a call fetching the users profile.
      */
-    constructor(response?: any) {
-        if (response) {
-            if (typeof response.type === "string") {
-                this.type = response.type;
+    constructor(response?: unknown) {
+        if (typeof response === "object" && response !== null) {
+            const { type, identities, data } = response as {
+                [k: string]: unknown;
+            };
+
+            if (typeof type === "string") {
+                this.type = type as UserType;
             } else {
                 throw new Error("Expected 'type' in the response body");
             }
 
-            if (Array.isArray(response.identities)) {
-                this.identities = response.identities.map((identity: any) => {
+            if (Array.isArray(identities)) {
+                this.identities = identities.map((identity: any) => {
                     return {
                         id: identity.id,
-                        providerId: identity["provider_id"],
+                        // providerId: identity["provider_id"],
```

Comment:
Should the line be removed?

Reply:
Probably 😊 I might have wanted to leave it in as a signal that this is infact also returned from the API, but it should probably be removed.